internal/ui/settings_ui.go

```diff
@@ -361,3 +370,29 @@ func modifierToString(mods deskDriver.Modifier) string {
 	}
 	return strings.Join(s, "+")
 }
+
+func getPicturesDir() (fyne.ListableURI, error) {
+	home, err := os.UserHomeDir()
+	if err != nil {
+		return nil, err
+	}
+
+	const xdg = "xdg-user-dir"
+	if _, err := exec.LookPath(xdg); err == nil {
+		cmd := exec.Command(xdg, "PICTURES")
+
+		loc, err := cmd.Output()
+		if err == nil {
```

Comment:
I think this previous comment is still needing addressed:
 
> Note lastly that xdg-user-dir will fall back to the home directory, so should check to see if that is the case and fall back to a better default

You'll need to check whether or not the returned location is the home directory (like https://github.com/fyne-io/fyne/blob/0ca05dc12c03b8873d2f22caa6ce4c28988b3886/dialog/file_xdg.go#L36).

Reply:
Good point. Changed it so it will try the fallback if that is the case.